Sources/NIOSSL/SSLPrivateKey.swift

```diff
@@ -266,6 +277,18 @@ public class NIOSSLPrivateKey {
         try self.init(bytes: bytes, format: format, callbackManager: manager)
     }
 
+    /// Create a `NIOSSLPrivateKey` from a custom private key callback.
+    ///
+    /// The private key, in addition to needing to conform to `NIOSSLCustomPrivateKey`,
+    /// is also required to be `Hashable`. This is because `NIOSSLPrivateKey`s are `Hashable`.
+    ///
+    /// - parameters:
+    ///     - customPrivateKey: The custom private key to use with the TLS certificate.
+    @inlinable
+    public init<CustomKey: NIOSSLCustomPrivateKey & Hashable>(customPrivateKey: CustomKey) {
+        self.representation = .custom(AnyNIOSSLCustomPrivateKey(customPrivateKey))
```

Comment:
Just a question; would this be the entry point for any key that is created outside the context of BoringSSL's `EVP_PKEY` type?

Reply:
Yup